src/client/l10n/useCurrentLocale.ts

```diff
@@ -0,0 +1,26 @@
+import { matchPath, useLocation } from 'react-router-dom'
+import { LocaleLabel, locales } from './locales'
+import { localePathPattern } from './localePathPattern'
+
+type LocaleUrlParam = {
+  locale: LocaleLabel
+}
+
+const FALLBACK_LOCALE: LocaleLabel = 'se-en'
+
+const getLocaleFromPath = (path: string): LocaleLabel => {
+  const localeMatch = matchPath<LocaleUrlParam>(path, {
+    path: localePathPattern + '/*',
+  })
+
+  const localeFromPath = localeMatch?.params?.locale
```

Comment:
Do we need to make it lowercase like we had it before or is that redundant? 
```suggestion
  const localeFromPath = localeMatch?.params?.locale?.toLowerCase()
```

Reply:
We don't since it's typesafe now and we can know it will only be exactly the same strings as the `LocaleLabel` union type 😇

Reply:
Actually, seems like we should do a `.toLowerCase()` in order to allow `/Se-EN` and other crazy versions of the locale params (which I think we _should_ allow and we also do today). But it has to be done earlier because at this stage there can only be lower case strings (because of the type awesomeness).

So I added so we're making  entire `location.pathname` lower case before passing it as an argument to `getLocaleFromPath`


![image](https://user-images.githubusercontent.com/42962286/134477351-f09d894d-8264-46a0-b118-888634975a85.png)

Reply:
Awesome, good catch! 🙏

Reply:
Well it's all thanks to you 😊